### PR TITLE
[ToDo]: move is_affiliated()

### PR DESCRIPTION
**Pull request by raj454raj**

### CODE REVIEW

File: modules/s3menus.py [hunk 1]
```diff
@@ -1439,7 +1439,7 @@ def org():
 
         return M(c="org")(
                     M("Organizations", f="organisation")(
-                        M("Create Organization", m="create"),
+                        M("Create", m="create"),
```
Comment (raj454raj) [inline modules/s3menus.py]: Just to maintain uniformity in all the modules.

File: modules/s3/s3aaa.py [hunk 2]
```diff
@@ -5328,6 +5328,28 @@ def get_owners(self, table, record):
             owner_user = record["owned_by_user"]
         return (realm_entity, owner_group, owner_user)
 
+    # -----------------------------------------------------------------------------
+    def is_affiliated(self):
```
Comment (nursix) [inline modules/s3/s3aaa.py]: definitely the wrong place - nothing to do whatsoever with permissions.

Why move it at all? It is only used in req and nowhere else, so it seems specific. And even if not - it should either go into org (most logical to me) or into AuthS3 (still logical), but certainly not S3Permission.

Comment (nursix) [inline modules/s3/s3aaa.py]: To be honest, I would leave this in req, and remove the todo. It is only used in the req controller and nowhere else - moving it into a different place will only make debugging harder for no reason.

Comment (raj454raj) [inline modules/s3/s3aaa.py]: Yeah didn't made sense to me either but as the Todo suggested. Okay I'll remove Todo.

**Comment by nursix** [comment]
ok :+1:

**Comment by nursix** [comment]
Sometimes (I said that before) a ToDo is a sign for an immature idea - and hence it should make sense to you before you action it. The general rule in open source programming is "change the code if you can improve it" - not change it because you can change it ;)

**Comment by raj454raj** [comment]
Sorry for that :p , I actually thought I'll ask flavour for that on IRC but he wasn't around.

**Comment by nursix** [comment]
I am a bit uncertain about the is_affiliated-extension of the business rule here: what does it mean? Do we only check permitted facilities if the user is affiliated with an organisation? Can't quite make sense of it yet. An explanation would be nice.

My feeling is that there must have been a reason why it had not been done before despite available - if it would have been /that/ simple, then it wouldn't have been a ToDo.

**Comment by raj454raj** [comment]
Okay one doubt: mpr module doesnt have its table definitions in s3db, is it not intended to make one or the module's not in use.

**Comment by nursix** [comment]
mpr is an extra perspective of dvi, not a module of its own.

**Comment by raj454raj** [comment]
Ok. :+1:

**Comment by nursix** [comment]
i.e. it doesn't have its own model, but uses pr and dvi models.

Generally, you can consider controller files as "perspectives" (or use-cases) rather than as "modules" - Eden isn't really modular in that sense. Some models have multiple perspectives, e.g. doc and gis are ubiquitous, supply is used both for req and inv (but has no independent use-case), stats is a meta-model, as is sit. 

And there are also perspectives which extend others, and do not necessarily need any additional model of their own: mpr extends dvi (no extra model needed), vol extends hrm (minimal extra model), ...

**Comment by raj454raj** [comment]
Thanks.

**Comment by nursix** [comment]
NB Are you GCI contestant, or Sahana volunteer? Just wondering why you're trying to action arbitrary ToDo's.

**Comment by raj454raj** [comment]
Sahana Volunteer.

File: controllers/req.py [hunk 3]
```diff
@@ -468,9 +467,10 @@ def prep(r):
             elif type == 3: # People
                 # Limit site_id to orgs the user has permissions for
                 # @ToDo: Make this customisable between Site/Org
-                # @ToDo: is_affiliated()
-                auth.permitted_facilities(table=r.table,
-                                          error_msg=T("You do not have permission for any facility to make a commitment."))
+                if is_affiliated():
```
Comment (flavour) [inline controllers/req.py]: I don't think this is what is meant by the ToDo...just hiding the permission check isn't valuable.
I think the issue here is that *once Commitments can be made by an Org rather than a Facility* then is_affiliated can be used to determine whether a user has the ability to commit on behalf of an org or not
The first part is the harder part since the Requests module is currently tied to all Requests/Commitments happening at the Site level...in order to support Inventory/Shipments...but these don't apply as much to Person commitements (people will often travel directly from their homes, even if to a staging area before actual deployment. The staging area is only sometimes their office)

File: modules/s3menus.py [hunk 4]
```diff
@@ -1439,7 +1439,7 @@ def org():
 
         return M(c="org")(
                     M("Organizations", f="organisation")(
-                        M("Create Organization", m="create"),
```
Comment (flavour) [inline modules/s3menus.py]: +1 to this for consistency.
Some templates may decide to be explicit about the create links, but it should be consistent either way.
(Personally I like to reduce the 'noise' of redundant information, but this does depend to some extent on the menu styling & structure as to how explicit we need to be here)

Comment (flavour) [inline modules/s3menus.py]: Done this in another commit...too small to justify a PR ;)

Comment (raj454raj) [inline modules/s3menus.py]: Sure thanks :+1:
